### ISSUE TITLE
Descendant refs can be duplicated on `.move()`

### ISSUE THREAD

**Issue by metincansiper** [cytoscape/cytoscape.js#1571]
I realized that after 'cy.move()' children information is not updated sometimes. Please see the following fiddle ( http://jsbin.com/bayugunusa/1/edit?js,console,output ). In this fiddle number of children that 'd' has seems to be 1 though it is expected to be 1. If you try to remove 'd' by enabling the latest code line which is currently commented out you can see that 'a' is removed as well though 'd' is not the parent of 'a'. (I think that this is caused by that 'a' still seems to be a child of 'd' unexpectedly)

**Comment by maxkfranz** [comment]
@metincansiper Would you be able to test this off of the unstable or master branch?  (Whichever you prefer)

**Comment by metincansiper** [comment]
@maxkfranz I tried to rebuild (by running 'gulp' command I tried 'gulp build' command as well) and test Cytoscape.js in both unstable and master branches, but in both cases 'src/version.js' is created but 'cytoscape.js' is not changed (Actually I was expecting that the change in your commits will be reflected to 'cytoscape.js' file). However, I tried to reflect the changes to the 'Cytoscape.js' file that I cloned to test this issue manually. It seems that the problem is fixed with your commit in this issue, thanks.
